visible.lua

```diff
@@ -0,0 +1,156 @@
+local LIBRARY_PATH = (...):match("(.-)[^%.]+$")
+local vector = require(LIBRARY_PATH .. "vector")
+local visible = {}
+
+
+local function comp(a, b, center)
+  if a.start == b.start then
+    if a.stop == nil then return false end
+    if b.stop == nil then return true end
+    return vector.alignment(a.start, a.stop, b.stop) > 0
+  end
+  return vector.polar_lt(a.start, b.start, center)
+end
+
+function visible.polygon(segments, center)
+  -- generic visibility function
+  -- return concave (frequently) polygon
+  if center == nil then center = vector.zero end
+  local endpoints = {}
+  do
```

Comment:
This do seems to just form a block for no reason. I didn't actually know you could put a do statement alone without causing a syntaxerror.